src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIPacket.cs

```diff
@@ -20,13 +21,17 @@ internal class SNIPacket : IDisposable, IEquatable<SNIPacket>
         private string _description;
         private SNIAsyncCallback _completionCallback;
 
-        /// <summary>
-        /// Constructor
-        /// </summary>
-        /// <param name="handle">Owning SNI handle</param>
-        public SNIPacket(SNIHandle handle)
+        private ArrayPool<byte>  _arrayPool = ArrayPool<byte>.Shared;
+        private bool _isBufferFromArrayPool = false;
+
+        public SNIPacket()
         {
-            _offset = 0;
+            _capacity = -1;
```

Comment:
Why can't this be 0 instead of -1 ?

Reply:
capacity = 0 means `_data = new byte[0];`
capacity = -1 means `_data = null;`

Reply:
It doesnt make sense to overload the value of negative to signify what _data looks like. Can't you simply check if _data = null instead of relying on a negative value of capacity?